google/resource_google_project_service.go

```diff
@@ -67,6 +67,16 @@ func resourceGoogleProjectServiceRead(d *schema.ResourceData, meta interface{})
 		return err
 	}
 
+	project, err := config.clientResourceManager.Projects.Get(id.project).Do()
+	if err != nil {
+		return handleNotFoundError(err, d, id.project)
+	}
+	if project.LifecycleState == "DELETE_REQUESTED" {
+		log.Printf("[WARN] Removing %s from state, its project is deleted", id.terraformId())
```

Comment:
Having this in delete makes lots of sense, but having it in READ makes me slightly uncomfortable.